Tests/FluentAssertions.Specs/Collections/Data/DataColumnCollectionAssertionExtensionsSpecs.cs

```diff
@@ -0,0 +1,433 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using FluentAssertions.Execution;
+
+using Xunit;
+using Xunit.Sdk;
+
+namespace FluentAssertions.Specs.Collections.Data
+{
+    public class DataColumnCollectionAssertionExtensionsSpecs
+    {
+        private DataColumn CreateTestDataColumn(int seed)
+        {
+            var column = new DataColumn("Column" + seed);
+
+            var random = new Random(seed);
+
+            column.DataType =
+                random.Next(4) switch
+                {
+                    0 => typeof(int),
+                    1 => typeof(string),
+                    2 => typeof(bool),
+                    _ => typeof(DateTime),
+                };
+
+            return column;
+        }
+
+        #region BeSameAs & NotBeSameAs
+        [Fact]
+        public void When_testing_that_references_to_the_same_object_are_the_same_it_should_succeed()
+        {
+            // Arrange
+            var dataTable = new DataTable("Test");
+
+            var columnCollection1 = dataTable.Columns;
+            var columnCollection2 = columnCollection1;
+
+            // Act & Assert
+            columnCollection1.Should().BeSameAs(columnCollection2);
+        }
+
+        [Fact]
+        public void When_testing_that_references_to_the_same_object_are_not_the_same_it_should_fail()
+        {
+            // Arrange
+            var dataTable = new DataTable("Test");
+
+            var columnCollection1 = dataTable.Columns;
+            var columnCollection2 = columnCollection1;
+
+            // Act
+            Action action =
+                () => columnCollection1.Should().NotBeSameAs(columnCollection2);
+
+            // Assert
+            action.Should().Throw<XunitException>();
```

Comment:
❌ Always ensure that the exception was thrown for the right reasons using `WithMessage` and wildcards.